### ISSUE TITLE
gax-java does not convert native gRPC exceptions in the server streaming case

### ISSUE THREAD

**Issue by kmjung** [GoogleCloudPlatform/google-cloud-java#2671]
I have a gRPC-based service consisting of a unary API and a server streaming API:

```
service myService {
  rpc UnaryMethod(UnaryRequest) returns (UnaryResponse) {
  };

  rpc ServerStreamingMethod(ServerStreamingRequest) returns (stream ServerStreamingResponse) {
  };
}
```

When the server returns a `google.rpc.Status` error, the gRPC runtime raises an `io.grpc.StatusException` in both cases. In the unary case, the exception is translated to a `com.google.api.gax.rpc.ApiException` inside `com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture`, but in the server streaming case, the `io.grpc.StatusException` is surfaced directly. I believe the behavior should be consistent between these two cases (and presumably in the client and bidirectional streaming cases as well).

**Comment by garrettjonesgoogle** [comment]
I believe ongoing work by @igorbernstein2 will fix this.

**Comment by igorbernstein2** [comment]
Yep, I'll update this ticket as soon as the relevant parts of https://github.com/googleapis/gax-java/pull/433 get merged

**Comment by igorbernstein2** [comment]
FYI, the changes have been merged and will be available in the next gax release

**Comment by pongad** [comment]
Fixed by #2825
